From c30e66e0fe9544a8088689bd1a0081f45ec44190 Mon Sep 17 00:00:00 2001
From: Chen Meng J <meng.j.chen@intel.com>
Date: Tue, 21 Aug 2018 10:18:57 +0800
Subject: [PATCH 34/54] media: intel-ipu4: ti960: pdata for TI960

Message for Open Source:
pdata for TI960

Message for Internal:
[Feature] Deserializer TI960 driver enabling
[Root Cause/Changes] pdata for TI960

Change-Id: If1670e065c72c327a406dea038face686565b7ab
Tracked-On: #JIIAP-573
Signed-off-by: Leo Zhao <leo.zhao@intel.com>
Signed-off-by: Chen Meng J <meng.j.chen@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/platform/intel/ipu4-bxt-p-pdata.c | 65 +++++++++++++++++++++++++
 1 file changed, 65 insertions(+)

diff --git a/drivers/media/platform/intel/ipu4-bxt-p-pdata.c b/drivers/media/platform/intel/ipu4-bxt-p-pdata.c
index f7b894e..5c43002 100644
--- a/drivers/media/platform/intel/ipu4-bxt-p-pdata.c
+++ b/drivers/media/platform/intel/ipu4-bxt-p-pdata.c
@@ -10,6 +10,7 @@
 #include <media/ipu-isys.h>
 #include <media/crlmodule.h>
 #include <media/ti964.h>
+#include <media/ti960.h>
 #include <media/max9286.h>
 #include "ipu.h"
 #include <media/dw9714.h>
@@ -988,6 +989,66 @@
 };
 #endif
 
+#if IS_ENABLED(CONFIG_VIDEO_TI960)
+#define TI960_I2C_ADAPTER	2
+#define TI960_I2C_ADAPTER_2	7
+#define TI960_LANES	4
+
+static struct ipu_isys_csi2_config ti960_csi2_cfg = {
+	.nlanes = TI960_LANES,
+	.port = 0,
+};
+
+static struct ipu_isys_csi2_config ti960_csi2_cfg_2 = {
+	.nlanes = TI960_LANES,
+	.port = 4,
+};
+
+static struct ti960_subdev_info ti960_subdevs[] = {
+};
+
+static struct ti960_subdev_info ti960_subdevs_2[] = {
+};
+
+static struct ti960_pdata ti960_pdata = {
+	.subdev_info = ti960_subdevs,
+	.subdev_num = ARRAY_SIZE(ti960_subdevs),
+	.reset_gpio = GPIO_BASE + 63,
+	.suffix = 'a',
+};
+
+static struct ipu_isys_subdev_info ti960_sd = {
+	.csi2 = &ti960_csi2_cfg,
+	.i2c = {
+		.board_info = {
+			 .type = "ti960",
+			 .addr = TI960_I2C_ADDRESS,
+			 .platform_data = &ti960_pdata,
+		},
+		.i2c_adapter_id = TI960_I2C_ADAPTER,
+	}
+};
+
+static struct ti960_pdata ti960_pdata_2 = {
+	.subdev_info = ti960_subdevs_2,
+	.subdev_num = ARRAY_SIZE(ti960_subdevs_2),
+	.reset_gpio = GPIO_BASE + 66,
+	.suffix = 'b',
+};
+
+static struct ipu_isys_subdev_info ti960_sd_2 = {
+	.csi2 = &ti960_csi2_cfg_2,
+	.i2c = {
+		.board_info = {
+			 .type = "ti960",
+			 .addr = TI960_I2C_ADDRESS,
+			 .platform_data = &ti960_pdata_2,
+		},
+		.i2c_adapter_id = TI960_I2C_ADAPTER_2,
+	}
+};
+#endif
+
 #ifdef CONFIG_INTEL_IPU4_AR0231AT
 #define AR0231AT_LANES            4
 #define AR0231ATA_I2C_ADDRESS      0x11
@@ -1150,6 +1211,10 @@
 		&ti964_sd,
 		&ti964_sd_2,
 #endif
+#if IS_ENABLED(CONFIG_VIDEO_TI960)
+		&ti960_sd,
+		&ti960_sd_2,
+#endif
 #ifdef CONFIG_INTEL_IPU4_OV2775
 		&ov2775_crl_sd,
 #endif
-- 
1.9.1

